### PR TITLE
CSG-22 - Create API doc

### PR DESCRIPTION
**Pull request by diegodalmolini**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+```
+---
+## Summary
+- [Players](#players)
+  - [Register players](#register-players)
+- [Matches](#matches)
+  - [Register matches](#register-matches)
+- [Ranks](#ranks)
+  - [Kills](#rank-kills)
+    - [Display all kills sorted by higher to lower](#rank-kills-sorted)
```
Comment (marcelosj3) [inline README.md:14]: This link leads to nowhere, maybe you can delete this and implement the errors and/or the query params in these cases.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+{
+	"url": "https://csgostats.gg/player/76561198062653475"
+}
+```
+Response:
+```
+HTTP: 200 - OK
+```
+```json
+{
```
Comment (marcelosj3) [inline README.md:38]: probably no big deal, but could we do 2 spacings, i think these jsons are really spaced, something like this:
```json
{
  "playerId": "850fde80-8231-452t-b39c-f9b23695385a",
  "player": "SK|Blavold",
  "imageUrl": "https://avatars.akamai.steamstatic.com/e66064d13f75fe43da64fce740ac29a217f9fd6a_full.jpg",
  "platformCredentials": [
    {
      "platformName": "CSGOSTATS",
      "platformPlayerId": "76561198062653475"
    }
  ]
}
```

File: README.md [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+	"code": 400,
+	"message": {
+		"error": "Invalid url received"
+	}
+}
+```
+</details>
+
+---
+## <center>**Matches** <a name="cities"></a></center>
```
Comment (marcelosj3) [inline README.md:76]: We have no id named `cities`, probably wanted to put `matches` here?

File: README.md [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+---
+## <center>**Ranks** <a name="ranks"></a></center>
+---
+### Kills <a name="rank-kills"></a>
+
+```
+GET - /api/ranks/kills/
+```
+
+Display the kill rank of all registered players sorted by higher to lower and the specific match's url
```
Comment (marcelosj3) [inline README.md:170]: Put only `Display the number of kills from all the registered players sorted from the higher to the lower amount` in here, or something like this, feel free to make it better, we need to specify that the user is obliged to place a `?match_url` query params in order to display the specifc match url

File: README.md [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+		"kills": 32,
+		"matchUrl": "https://csgostats.gg/match/58067113"
+	},
+	{
+		"name": "SK|Blavold",
+		"kills": 30,
+		"matchUrl": "https://csgostats.gg/match/73434211"
+	}
+]
+```
```
Comment (marcelosj3) [inline README.md:214]: At the end of this one, you could specify, just like the `Errors` in the others, a section named `Query params`, where the user can place some query params, now we only have the `match_url` one, and display a response showing (perhaps not all of them) the kills rank with the `matchUrl`.

File: README.md [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+Response:
+```
+HTTP: 200 - OK
+```
+```json
+[
+	{
+		"name": "SK|Blavold",
+		"kills": 40,
+		"matchUrl": "https://csgostats.gg/match/72090725"
```
Comment (marcelosj3) [inline README.md:181]: Remove the `matchUrl` from all of the objects here, if the users makes a request like above (which is `GET - /api/ranks/kills/`), he will not get the `matchUrl`, only by placing the query params this value will be shown, that is `GET - /api/ranks/kills?match_url/`

**Review by marcelosj3 — CHANGES_REQUESTED** [review]
By the way, here you can find the commits pattern i am using for this project, maybe we can find a way to rewrite your commits into this https://www.conventionalcommits.org/en/v1.0.0/

File: README.md (hunk 2 above)
Comment (diegodalmolini) [reply README.md:49]: done

File: README.md (hunk 5 above)
Comment (diegodalmolini) [reply README.md:226]: done

**Review by marcelosj3 — CHANGES_REQUESTED** [review]

File: README.md [hunk 7]
```diff
@@ -0,0 +1,283 @@
...
+	}
+}
+```
+</details>
+
+---
+## <center>**Ranks** <a id="ranks"></a></center>
+---
+
+###Kills
```
[the remark this answers is not in the document]
Comment (diegodalmolini) [reply README.md:180]: done

File: README.md [hunk 8]
```diff
@@ -0,0 +1,283 @@
...
+    "kills": 30,
+  }
+]
+```
+</details>
+
+### Query params
+
+<details>
+<summary>Display all kills with match url</summary> <a id="rank-kills-match_url"></a>
```
[the remark this answers is not in the document]
Comment (diegodalmolini) [reply README.md:232]: maybe is done

**Pull request closed without merging**